applications/RANSApplication/custom_conditions/vms_monolithic_k_based_wall_condition.cpp

```diff
@@ -0,0 +1,200 @@
+//    |  /           |
+//    ' /   __| _` | __|  _ \   __|
+//    . \  |   (   | |   (   |\__ `
+//   _|\_\_|  \__,_|\__|\___/ ____/
+//                   Multi-Physics
+//
+//  License:         BSD License
+//                   Kratos default license: kratos/license.txt
+//
+//  Main authors:    Suneth Warnakulasuriya
+//
+
+// System includes
+#include <cmath>
+#include <functional>
+#include <iostream>
+#include <limits>
+#include <string>
+
+// External includes
+
+// Project includes
+#include "includes/checks.h"
+#include "includes/define.h"
+#include "includes/variables.h"
+
+// Application includes
+#include "custom_utilities/rans_calculation_utilities.h"
+#include "rans_application_variables.h"
+
+// Include base h
+#include "vms_monolithic_k_based_wall_condition.h"
+
+namespace Kratos
+{
+template <unsigned int TDim, unsigned int TNumNodes>
+int VMSMonolithicKBasedWallCondition<TDim, TNumNodes>::Check(
+    const ProcessInfo& rCurrentProcessInfo)
+{
+    KRATOS_TRY;
+
+    int check = BaseType::Check(rCurrentProcessInfo);
+
+    const auto& r_geometry = this->GetGeometry();
+
+    for (IndexType i_node = 0; i_node < TNumNodes; ++i_node) {
+        const auto& r_node = r_geometry[i_node];
+
+        KRATOS_CHECK_VARIABLE_IN_NODAL_DATA(TURBULENT_KINETIC_ENERGY, r_node);
+        KRATOS_CHECK_VARIABLE_IN_NODAL_DATA(DENSITY, r_node);
```

Comment:
This is intended to be used in an incompressible frameworks meaning that the nodal `DENSITY` is no longer required. Instead it must be retrieved from the parent element properties. Note that #7298 is merged, this operation would be always possible by default.

Reply:
As discussed, I will move everything to element and condition properties in a seperate PR :)